apps/explorer/lib/tx.ts

```diff
@@ -0,0 +1,85 @@
+import BigNumber from 'bignumber.js'
+import {
+  ExplorerTransaction,
+  ExplorerV2Transaction,
+} from '@siafoundation/explored-types'
+
+export type AddressSummary = {
+  address: string
+  sc?: BigNumber
+  sf?: number
+}
+
+export function getTransactionSummary(
+  transaction: ExplorerTransaction | ExplorerV2Transaction
+): AddressSummary[] {
+  const addressAmountMap = new Map<string, { sc?: BigNumber; sf?: number }>()
+
+  const updateAmount = (
+    key: string,
+    field: 'sc' | 'sf',
+    amount: BigNumber | number
+  ) => {
+    const entry = addressAmountMap.get(key) || {}
+    if (field === 'sc') {
+      const value = amount as BigNumber
+      entry.sc = (entry.sc || new BigNumber(0)).plus(value)
+    } else {
+      const value = amount as number
+      entry.sf = (entry.sf || 0) + value
+    }
+    addressAmountMap.set(key, entry)
+  }
+
+  // Subtract inputs.
+  transaction.siacoinInputs?.forEach((input) => {
+    const address =
+      'parent' in input ? input.parent.siacoinOutput.address : input.address
+    const value = new BigNumber(
+      'parent' in input ? input.parent.siacoinOutput.value : input.value
+    )
+    updateAmount(`${address}|sc`, 'sc', value.negated())
+  })
+
+  transaction.siafundInputs?.forEach((input) => {
+    const address =
+      'parent' in input ? input.parent.siafundOutput.address : input.address
+    const value = Number(
+      'parent' in input ? input.parent.siafundOutput.value : input.value
+    )
+    updateAmount(`${address}|sf`, 'sf', -value)
+  })
+
+  // Add outputs.
+  transaction.siacoinOutputs?.forEach((output) => {
+    const address = output.siacoinOutput.address
+    const value = new BigNumber(output.siacoinOutput.value)
+    updateAmount(`${address}|sc`, 'sc', value)
+  })
+
+  transaction.siafundOutputs?.forEach((output) => {
+    const address = output.siafundOutput.address
+    const value = Number(output.siafundOutput.value)
+    updateAmount(`${address}|sf`, 'sf', value)
+  })
+
+  // Create the array and sort sf then sc, both descending.
+  return Array.from(addressAmountMap.entries())
+    .map(([key, { sc, sf }]) => {
+      const [address, asset] = key.split('|') as [string, 'sc' | 'sf']
```

Comment:
Rather than string splitting keys, would it not be more clear to have separate sc and sf maps?